### PR TITLE
[server] allow fetching task specific prebuild-logs

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

Expose Prebuild Logs per task.

Screenshot 2024-05-30 at 16 22 05

## Related Issue(s)

See https://linear.app/gitpod/issue/ENT-172/prebuilds-show-logs-of-all-tasks

## How to test

Join my org https://se-prebuild-by-task.preview.gitpod-dev.com/orgs/join?inviteId=07ebd814-a02f-4aeb-abf3-c87f351bec47

Check [the prebuild details](https://se-prebuild-by-task.preview.gitpod-dev.com/prebuilds/ab554c76-8fc8-41fa-99ed-6017908aa9b5) and open the network tab in chrome devtools.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
        <li><b>🏷️ Name</b> - se-prebuild-by-task</li>
        <li><b>🔗 URL</b> - <a href="https://se-prebuild-by-task.preview.gitpod-dev.com/workspaces" target="_blank">se-prebuild-by-task.preview.gitpod-dev.com/workspaces</a>.</li>
        <li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
        <li><b>📦 Version</b> - </li>
        <li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-se-prebuild-by-task%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/gitpod-db/src/typeorm/workspace-db-impl.ts [hunk 1]
```diff
@@ -1014,7 +1015,7 @@ export class TypeORMWorkspaceDBImpl extends TransactionalDBImpl<WorkspaceDB> imp
         }
 
         const res = await query.getMany();
-        return res;
+        return res as PrebuiltWorkspaceWithWorkspace[];
```
Comment (geropl) [inline components/gitpod-db/src/typeorm/workspace-db-impl.ts:1018]: @svenefftinge Did you check if `res.workspace` is filled? I don't recall from the top of my head how/where `innerJoinAndMapOne` puts the workspace into the result shape.

File: components/gitpod-protocol/src/teams-projects-protocol.ts [hunk 2]
```diff
@@ -148,6 +148,7 @@ export interface ProjectUsage {
 export interface PrebuildWithStatus {
     info: PrebuildInfo;
     status: PrebuiltWorkspaceState;
+    workspace: Workspace;
```
Comment (geropl) [inline components/gitpod-protocol/src/teams-projects-protocol.ts:151]: :dart:

**Review by geropl — APPROVED** [review]
Backend code changes LGTM, tested streaming (without taskId) and it still works :heavy_check_mark:

The build issue is just asking for updating golden files

File: components/gitpod-db/src/typeorm/workspace-db-impl.ts (hunk 1 above)
Comment (svenefftinge) [reply components/gitpod-db/src/typeorm/workspace-db-impl.ts:1018]: yes, I added a unit test

**Pull request merged**